docs/standard/native-interop/disabled-marshaling.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Disabled runtime marshalling - .NET
+description: Learn how .NET marshals your types to a native representation when runtime marshalling is disabled.
+ms.date: 01/12/2022
+---
+
+# Disabled runtime marshalling
+
+When the [`System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute`](xref:System.Runtime.CompilerServices.DisableRuntimeMarshallingAttribute) attribute is applied to an assembly, the runtime disables most built-in support for data marshalling between managed and native representations. This document describes the features that are disabled and how .NET types map to native types when marshalling is disabled.
+
+## Scenarios where marshalling is disabled
+
+When the `DisableRuntimeMarshallingAttribute` is applied to an assembly, it affects P/Invokes and Delegate types in the assembly, as well as any calls to unmanaged function pointers in the assembly. It does not affect any P/Invoke or interop delegate types defined in other assemblies. It also does not disable marshalling for the runtime's built-in COM interop support. The built-in COM interop support can be enabled or disabled through a [feature switch](https://github.com/dotnet/runtime/blob/main/docs/workflow/trimming/feature-switches.md).
```

Comment:
Could you give some clarifying examples here?

It's not exactly clear which of the below work vs not and what the behavior is given
```csharp
[UnmanagedFunctionPointer(...)]
public delegate void Callback();

[DllImport(...)]
public static extern void Import(Callback callback);
```

Where you have the varying combinations of `Import` defined in `A` and `Callback` defined in `B` where `A` and/or `B` have `DisableRuntimeMarshalling` defined.

Reply:
-- It's also "somewhat" unfortunate that `COM interop` is controlled by a separate switch I think. It would be nice to have a "one stop shop" to disable all built-in marshalling and enforce blittable conventions.

It's not a big deal though, it just means one extra thing to tweak in my own libraries.

Reply:
Since this feature doesn't totally disable P/Invoke and delegate-based interop, it didn't make sense to fully disable COM interop as part of this attribute. Since COM would still have to generate IL stubs and use marshalling for the `this` parameter, it doesn't make sense (at least to me) to disable the marshalling but not the built-in COM system as a whole.

Reply:
> It would be nice to have a "one stop shop" to disable all built-in marshalling and enforce blittable conventions.

I'm not against that, but it seems hard in my mind. Does it mean the PIA itself &ndash; the containing assembly of the COM type &ndash; applies to that definition of the interface? Does it apply the GUID for the type? If the assembly using COM interop uses an interface does the attribute apply to the call? (I assume not). I guess we could also apply the attribute to the assembly that activates the COM object, but that also gets weird.

If you can come up with a crisp set of rules that makes intuitive sense I'd be very interested.